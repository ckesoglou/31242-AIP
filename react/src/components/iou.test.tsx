import React from "react";
import { shallow } from "enzyme";
import Iou from "./iou";

describe("<Iou />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(
      <Iou
        request={{
          id: "1",
          author: { username: "James", display_name: "James" },
          completed_by: { username: "Kevin", display_name: "Kevin" },
          proof_of_completion: "Some ID",
          rewards: [
            { id: "1", display_name: "Hug" },
            { id: "2", display_name: "Coffee" },
          ],
          details: "Clean the fridge",
          created_time: "02/02/2020",
          comletion_time: "02/02/2020",
          is_completed: true,
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
