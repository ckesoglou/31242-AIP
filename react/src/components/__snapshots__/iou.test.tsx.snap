// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Iou /> should render correctly 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={1}
    xs={12}
  >
    <WithStyles(ForwardRef(Grid))
      id="requestItemContainer"
      item={true}
      xs={6}
    >
      <IouFavourSingle
        giverDisplayName="James"
        item={
          Object {
            "display_name": "Hug",
            "id": "1",
          }
        }
        recieverDisplayName="?"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      id="requestItemContainer"
      item={true}
      xs={1}
    >
      <IouProof
        imagePK="Some ID"
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      id="requestProofContainer"
      item={true}
      xs={3}
    >
      <IouComplete
        request={
          Object {
            "author": Object {
              "display_name": "James",
              "username": "James",
            },
            "comletion_time": "02/02/2020",
            "completed_by": Object {
              "display_name": "Kevin",
              "username": "Kevin",
            },
            "created_time": "02/02/2020",
            "details": "Clean the fridge",
            "id": "1",
            "is_completed": true,
            "proof_of_completion": "Some ID",
            "rewards": Array [
              Object {
                "display_name": "Hug",
                "id": "1",
              },
              Object {
                "display_name": "Coffee",
                "id": "2",
              },
            ],
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      id="requestItemContainer"
      item={true}
      xs={1}
    >
      <RequestInfo
        request={
          Object {
            "author": Object {
              "display_name": "James",
              "username": "James",
            },
            "comletion_time": "02/02/2020",
            "completed_by": Object {
              "display_name": "Kevin",
              "username": "Kevin",
            },
            "created_time": "02/02/2020",
            "details": "Clean the fridge",
            "id": "1",
            "is_completed": true,
            "proof_of_completion": "Some ID",
            "rewards": Array [
              Object {
                "display_name": "Hug",
                "id": "1",
              },
              Object {
                "display_name": "Coffee",
                "id": "2",
              },
            ],
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
